[Zhong Baode] IP

ZhongBaode opened the pull request.

Commit:
Implemented an initial skeletal version

Commit:
Added list tracking feature

Mochi is now able to track and mark/unmark and thing from the list
#TODO
Switch to switch case
Add more features
Brush up the interaction

jyx0615 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,87 @@
+            System.exit(0);
+        }
+
+        if (input.trim().equalsIgnoreCase("list")) {
+            printArrayList(inputs);
+        }
+        else if (input.toLowerCase().contains("unmark")){
```
try to follow the coding standard that the `else if` should be in the same line as `}` with one space.
````
if (condition) {
    statements;
} else {
    statements;
}

jyx0615 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    private static void markAsUndone(String input, ArrayList<Task> inputs) {
+        try {
+            int listIndex = Integer.parseInt(input.substring(7)) - 1;
+            inputs.get(listIndex).markAsUndone();
+        }
+        catch (Exception e) {
```
try to follow the following coding standard
```
try {
    statements;
} catch (Exception exception) {
    statements;
}

jyx0615 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,87 @@
+    }
+
+    private static void markAsDone(String input, ArrayList<Task> inputs) {
+        try {
+            int listIndex = Integer.parseInt(input.substring(5)) - 1;
+            inputs.get(listIndex).markAsDone();
+        }
+        catch (Exception e) {
```
coding standard

jyx0615 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,87 @@
+        }
+        else if (input.toLowerCase().contains("unmark")){
+            markAsUndone(input, inputs);
+        }
+        else if (input.toLowerCase().contains("mark")){
```
if-else if statement coding standard

jyx0615 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,87 @@
+        }
+        else if (input.toLowerCase().contains("mark")){
+            markAsDone(input, inputs);
+        }
+
+        else{
```
if-else if statement coding standard

jyx0615 reviewed (commented):
try to follow the coding standard

Commit:
Updated main loop to include exit condition with isRunning

Commit:
Updated to switch cases instead of IF ELSE

nitin19011 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+public class Task {
+    private String description;
+    private boolean isDone = false;
```
good use of boolean variable naming

nitin19011 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,129 @@
+            inputs.get(listIndex).markAsUndone();
+        }
+        catch (Exception e) {
+            System.out.println("(“index out of range”, “please enter a number”");
```
Error message can be clearer along the lines of "please enter a valid task number"

nitin19011 commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,129 @@
+        }
+    }
+
+    private static void printArrayList(List<Task> inputs) {
```
Method name could be more informative like "printTasks"

nitin19011 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+        this.description = description;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
```
Method names are a little chunky, perhaps try "markDone"?

nitin19011 reviewed (commented):
Overall good use of OOP and a solid foundation to build the rest of the project on

Commit:
Added Deadline, event and Todo tasks to Mochi

Added deadline event and todo task capabilities to better track tasks
Updated If else loop to Switch cases too and cleaned up the code

Commit:
Refactored code

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+    private void runMain() {
+        welcomeMessage();
+        try (Scanner sc = new Scanner(System.in)) {
+            while(this.isRunning && sc.hasNextLine()){
```
Perhaps could add some whitespace for readability, this seems to be an issue in other areas of the code as well

NigelYeoTW replied:
Perhaps reduce the usage of 'this' keyword when not needed to remove redundancy

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+            case "unmark" ->    markAsUndone(input, taskList);
+            case "todo" ->      insertTodo(splits[DESCRIPTION_INDEX]);
+            case "event" ->     insertEvent(splits[DESCRIPTION_INDEX]);
+        }
```
Perhaps indent the case statements consistently and have the default case

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+
+    private static void insertEvent(String uncleanedString) {
+        String description = uncleanedString.substring(0, uncleanedString.indexOf(EVENT_CMD_FROM));
+        String startDate = uncleanedString.substring(uncleanedString.indexOf(EVENT_CMD_FROM) + EVENT_CMD_FROM.length(), uncleanedString.indexOf(EVENT_CMD_TO));
```
Perhaps add break lines when character count gets too high, this seem to be an issue in other areas of the code as well

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+    }
+    private static void markAsUndone(String input, List<Task> inputs) {
+        try {
+            int listIndex = Integer.parseInt(input.substring(7)) - 1;
```
Perhaps remove the magic numbers, this seems to be an issue in other areas of the code as well

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+        catch (Exception e) {
+            System.out.println("(“index out of range”, “please enter a number”");
+        }
+    }
```
Should the catch statement be on the following line?

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+        goodByeMessage();
+    }
+
+    private void handle(String raw){
```
Perhaps the function naming could be more intuitive

NigelYeoTW commented on src/main/java/Mochi.java:
```diff
@@ -0,0 +1,135 @@
+        addEventSuccess(task);
+    }
+
+    private  static void addEventSuccess(Task task){
```
Perhaps remove the extra spaces

NigelYeoTW reviewed (commented):
Overall lgtm, there just seems to be some minor formatting issues and minor coding standard violations, good job 👍

Commit:
Implemented Error Handling of Invalid Input and added Enums for Switch case

In the Enum class added a from string method to automatically convert, this helps identify correct error better, distinguishing empty description or wrong command

Commit:
Added more Exception Handling

Commit:
Packaged Tasks

Commit:
Added Read and Save file feature

Commit:
Added TODO comment

Commit:
Added Delete Command

Commit:
Renamed Save file to MochiData

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/mochi/Mochi.java

Commit:
Added textui testing code, cleaned up SOUT Code as well

Sout code had some strange characters so i cleaned it up

Commit:
Merge branch 'UITesting'

Commit:
Updated exception package to fit convention

Commit:
Added more OOP to the code

Commit:
Added find function

Used streams to find substring and overloaded print list function

Commit:
Added JavaDoc for classes

Commit:
Added Deleteall command

Commit:
Added Readme doc

Commit:
Bug fix Event and deadline

Commit:
Updated ui text input

Commit:
Added new FindTime feature

Find time will find all task related to the date and time

Commit:
Ignore save files

Commit:
Merge remote-tracking branch 'origin/master' into branch-Level-9

# Conflicts:
#	src/main/java/mochi/task/Commands.java

Commit:
Updated Javadoc for findtime and finditem

Commit:
Updated user guide

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Updated main README.file

Commit:
Updated main README.file

Commit:
Merge branch 'master' of github.com:ZhongBaode/ip

# Conflicts:
#	README.md

Commit:
Updated UItext tests
